alignment_check_experiment/01_umap_parameter_search.py

```diff
@@ -0,0 +1,243 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:biovectors]
+#     language: python
+#     name: conda-env-biovectors-py
+# ---
+
+# # Generate UMAP GIF for Token Evolution Progression
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+
+import itertools
+from pathlib import Path
+import pickle
+
+from adjustText import adjust_text
+import imageio
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import plotnine as p9
+import tqdm
+
+from umap.parametric_umap import ParametricUMAP, load_ParametricUMAP
```

Comment:
I haven't read the paper yet, is parametric UMAP just a faster way to do UMAP? Or does it have some other property that makes it nice?

Reply:
ParametricUMAP is basically UMAP meets neural networks. The beauty of this approach is that one doesn't run into memory issues and it allows for new datapoints to be embedding into the umap embedding (without retraining). According to the paper, it's an upgrade people were seeking to fix the embedding never-before seen datapoints issue.